### PR TITLE
Add the utility class for VTK.

### PR DESCRIPTION
**Pull request by kunimatsu-tri**
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/5200)

### CODE REVIEW

**Comment by kunimatsu-tri** [comment]
@patmarion feature review please.

FYI, I don't include color setting capability since I found that just setting color at actor level is the easiest way: http://www.vtk.org/Wiki/VTK/Examples/Cxx/Visualization/ColorAnActor

File: drake/systems/sensors/vtk_util.cc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  plane->SetPoint2(half_size, -half_size, 0.);
+  plane->Update();
+
+  return plane;
+}
+
+vtkSmartPointer<vtkTransform> VtkUtil::ConvertToVtkTransform(
+    const Eigen::Isometry3d& transform) {
+  auto position = transform.translation();
+  auto axis_angle = drake::math::rotmat2axis(transform.linear());
```
Comment (patmarion) [inline drake/systems/sensors/vtk_util.cc]: you might just want to set the matrix elements, as in https://github.com/RobotLocomotion/director/blob/master/src/app/ddDrakeModel.cpp#L130

**Comment by kunimatsu-tri** [comment]
+@patmarion

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-:-Kd8Txr0HMqVbVHItPbl:b-hrunxp)*

File: drake/systems/sensors/vtk_util.cc (hunk 1 above)
Comment (kunimatsu-tri) [inline drake/systems/sensors/vtk_util.cc]: Good to know. Thanks!

BTW, this is exactly the same function. Perhaps, I will need to consider just porting vtk utilities from director to Drake.

**Comment by kunimatsu-tri** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[drake/systems/sensors/vtk_util.cc, line 36 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-Kd7rDjqcxVK5X84swQa-r1-36:-Kd8vMW7-Y3W1H09bgvj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/e634eb4e6ae60680a6b21aa9eafa12523461824a/drake/systems/sensors/vtk_util.cc#L36)):*
<details><summary><i>Previously, kunimatsu-tri (Kunimatsu Hashimoto) wrote…</i></summary><blockquote>

Good to know. Thanks!

BTW, this is exactly the same function. Perhaps, I will need to consider just porting vtk utilities from director to Drake.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200)*

**Comment by patmarion** [comment]
:lgtm:

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-:-KdC-RpMxDIXa30Zq_Kt:bnfp4nl)*

**Comment by kunimatsu-tri** [comment]
+@jwnimmer-tri platform review please

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-:-KdC4758iMWVHT0lKNsU:b-k0crfs)*

**Comment by liangfok** [comment]
Reviewed 2 of 5 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200)*

**Comment by kunimatsu-tri** [comment]
@drake-jenkins-bot mac-clang-experimental please
@drake-jenkins-bot mac-clang-ninja-experimental please

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-:-KdC4oDXb_VBTVVBqxQY:bg0xcbm)*

**Comment by jwnimmer-tri** [comment]
Reviewed 2 of 5 files at r1, 2 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/systems/sensors/vtk_util.cc, line 4 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCGfxIOcxx-YiZywKm:-KdCGfxJxoA_muiFxFKd:b-wlijz4) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/vtk_util.cc#L4)):*
> ```cpp
> 
> #include <Eigen/Dense>
> ```

BTW Remove this blank line.  All "library" headers go in a single group.  In https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes they are all item "4." in the ranking.

---

*[drake/systems/sensors/vtk_util.h, line 4 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCGIG43g2rhr63vz9s:-KdCGIG43g2rhr63vz9t:bbxi8st) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/vtk_util.h#L4)):*
> ```objc
> 
> #include <Eigen/Dense>
> ```

BTW Remove this blank line.  All "library" headers go in a single group.

---

*[drake/systems/sensors/vtk_util.h, line 18 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCGPkjNEXZAllM034I:-KdCGPkjNEXZAllM034J:bjf9wlc) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/vtk_util.h#L18)):*
> ```objc
> class VtkUtil {
>  public:
>   DRAKE_NO_COPY_NO_MOVE_NO_ASSIGN(VtkUtil)
> ```

BTW Typically we would also delete the ctor, so the class is not instantiable at all.  But I think its discretionary, not required?

---

*[drake/systems/sensors/test/vtk_util_test.cc, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCK14cmuIPtzk-fK_v:-KdCK14cmuIPtzk-fK_w:b-prh9yh) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/test/vtk_util_test.cc#L13)):*
> ```cpp
> #include <vtkCellData.h>
> #include <vtkPlaneSource.h>
> #include <vtkSmartPointer.h>
> ```

BTW Put Eigen and VTK includes in one place; they are all "library" headers.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200)*

**Comment by jwnimmer-tri** [comment]
Reviewed 1 of 3 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/systems/sensors/test/vtk_util_test.cc, line 79 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCRKsb64-8cMmu8ijh:-KdCRKsb64-8cMmu8iji:bbjldic) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/test/vtk_util_test.cc#L79)):*
> ```cpp
>     dut->GetOutput()->GetPoints()->GetPoint(i, dut_point);
>     EXPECT_TRUE(MatchAndRemovePoint(dut_point));
>   }
> ```

The LOC count to check four simple points is very large.  Is the order of points returned by `CreateSquarePlane` actually non-deterministic?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200)*

**Comment by kunimatsu-tri** [comment]
Review status: 2 of 5 files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/systems/sensors/vtk_util.cc, line 4 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCGfxIOcxx-YiZywKm:-KdCU5NIutHsIJxzbDFT:b1r865v) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/vtk_util.cc#L4)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW Remove this blank line.  All "library" headers go in a single group.  In https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes they are all item "4." in the ranking.
</blockquote></details>

Done.

FYI, I put `#include <Eigen/Dense>` right below the VTK headers, so that it will be alphabet order like abc...xyzABC... 

---

*[drake/systems/sensors/vtk_util.h, line 4 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCGIG43g2rhr63vz9s:-KdCUPm4vHk2E_DwFYbJ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/vtk_util.h#L4)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW Remove this blank line.  All "library" headers go in a single group.
</blockquote></details>

Done.

---

*[drake/systems/sensors/vtk_util.h, line 18 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCGPkjNEXZAllM034I:-KdCWbzPzxqyudyN7N5O:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/vtk_util.h#L18)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW Typically we would also delete the ctor, so the class is not instantiable at all.  But I think its discretionary, not required?
</blockquote></details>

Done.

---

*[drake/systems/sensors/test/vtk_util_test.cc, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCK14cmuIPtzk-fK_v:-KdCVPIrvdADl9BfZPho:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/test/vtk_util_test.cc#L13)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW Put Eigen and VTK includes in one place; they are all "library" headers.
</blockquote></details>

Done.

---

*[drake/systems/sensors/test/vtk_util_test.cc, line 79 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-KdCRKsb64-8cMmu8ijh:-KdC_q2Qm38_9gkuPGtu:bw3rdpz) ([raw file](https://github.com/robotlocomotion/drake/blob/c9cbee31cdbba70ce152c9e83b5e7e309d84f770/drake/systems/sensors/test/vtk_util_test.cc#L79)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

The LOC count to check four simple points is very large.  Is the order of points returned by `CreateSquarePlane` actually non-deterministic?
</blockquote></details>

That's a good point! It was deterministic.
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200)*

**Comment by jwnimmer-tri** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5200#-:-KdCc8GyvC4uwVk06o0j:bnfp4nl)*

**Pull request merged**